src/basilisp/core.lpy

```diff
@@ -2757,13 +2758,15 @@
         (rf result (apply f input inputs))))))
   ([f coll]
    (lazy-seq
-    (when (seq coll)
+    (when-let [coll (seq coll)]
       (cons (f (first coll)) (map f (rest coll))))))
   ([f coll & colls]
    (lazy-seq
-    (when (and (seq coll) (every? seq colls))
-      (cons (apply f (first coll) (map first colls))
-            (apply map f (rest coll) (map rest colls)))))))
+    (when-let [coll (seq coll)]
+      (let [colls (map seq colls)]
+        (when (every? some? colls)
```

Comment:
Why `some?` rather than `identity`?

Reply:
Updated, thanks

I was reasoning at the semantics level, just checking that every value wasn't nil, so I used `(every? some?)`. I hadn't considered using `identity`, which relies on knowing that `(map seq colls)` returns either a `seq` or `nil`.

Reply:
ah, I missed the fact that the PR was already merged 😅